community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/storage/FailureStorage.java

```diff
@@ -115,6 +115,10 @@ public synchronized void storeIndexFailure( String failure ) throws IOException
         File failureFile = failureFile();
         try ( StoreChannel channel = fs.open( failureFile, "rw" ) )
         {
+            byte[] existingData = new byte[(int) channel.size()];
+            channel.read( ByteBuffer.wrap( existingData ) );
+            channel.position( lengthOf( existingData ) );
```

Comment:
Because a failure file is pre-allocated with all-zeros, so that a failure can be written even in an disk-space-full scenario. Sure, OK correct read (also in existing method `loadIndexFailure()` should be correct.